models/articles.model.js

```diff
@@ -24,6 +24,37 @@ exports.selectArticleById = (article_id) => {
     })
 }
 
+exports.insertComment = (commentData) => {
+    return db.query('SELECT * FROM articles WHERE article_id = $1', [commentData.article_id])
+    .then((articles) => {
+        if(!articles.rows[0]){
+            return Promise.reject({
+                status: 404,
+                msg: 'Not Found'
+            })
+        }
+    })
+    .then(() => {
+        return db.query('SELECT * FROM users WHERE username = $1', [commentData.username])
+        .then((users) => {
+            if(!users.rows[0]){
+                return Promise.reject({
+                    status: 404,
+                    msg: 'No Such User'
+                })
+            }
+        })
+    })
```

Comment:
I like your solution here but how does postgres handle these issues if you delete these checks? How can you use that to only make one query to the db?